rocketpy/EnvironmentAnalysis.py

```diff
@@ -2064,6 +2065,13 @@ def animate_average_wind_rose(self, figsize=(8, 8), filename="wind_rose.gif"):
         -------
         Image : ipywidgets.widgets.widget_media.Image
         """
+        try:
+            import ipywidgets as widgets
```

Comment:
I had to repeat this code block twice.
If you have an idea on how to avoid that, I would appreciate.
but it works.

Reply:
It is not that much repetition in a part of the code that is not crucial. I think it is fine to leave it like this

Reply:
agreed

Reply:
Could you update be on the reasoning behind moving the auxiliary imports to their own methods? My understanding is that one should only be able to create an `EnvironmentAnalisys` instance if all of its methods are readily available. Then, the imports would be on the top of the file, like:

```py
try:
    import ipywidgets, windrose, timezonefinder
except ImportError import_error:
    raise ImportError(f"Please install module {import_error}.")

### Please note that this is just a sketch code block as an example, I don't even know if it works.
```

I understand that the some of these imports are only needed sometimes in only one method. However, this is not the way the other imports are handled, for instance, there is no check for `matplotlib` at every `allInfo`, even though its the only place the plots module is needed.

Reply:
> Could you update be on the reasoning behind moving the auxiliary imports to their own methods? My understanding is that one should only be able to create an `EnvironmentAnalisys` instance if all of its methods are readily available. Then, the imports would be on the top of the file, like:
> 
> ```python
> try:
>     import ipywidgets, windrose, timezonefinder
> except ImportError import_error:
>     raise ImportError(f"Please install module {import_error}.")
> 
> ### Please note that this is just a sketch code block as an example, I don't even know if it works.
> ```
> 
> I understand that the some of these imports are only needed sometimes in only one method. However, this is not the way the other imports are handled, for instance, there is no check for `matplotlib` at every `allInfo`, even though its the only place the plots module is needed.

Great question,
I agree with your argument,
I guess I only tried to copy what had been done before with the timezonefinder packages, 
without actually thinking about what the code was doing.

Thanks for pointing out this kind of things.